src/Uno.UI/UI/Xaml/Controls/MenuFlyout/CascadingMenuHelper.cs

```diff
@@ -0,0 +1,977 @@
+
+			if (subMenuPresenterAsFE != null)
+			{
+				subMenuPresenter = subMenuPresenterAsFE as IMenuPresenter;
+			}
+
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: CloseChildSubMenus - subMenuPresenter=0x%p.", this, subMenuPresenter));
+#endif // CMH_DEBUG
+
+			if (subMenuPresenter != null)
+			{
+				subMenuPresenter.CloseSubMenu();
+			}
+		}
+
+		internal void DelayCloseSubMenu()
+		{
+			EnsureDelayCloseMenuTimer();
+			if (m_delayCloseMenuTimer != null)
+			{
+#if CMH_DEBUG
+				(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: DelayCloseSubMenu - Starting m_delayCloseMenuTimer.", this));
+#endif // CMH_DEBUG
+
+				m_delayCloseMenuTimer.Start();
+			}
+		}
+
+		internal void CancelCloseSubMenu()
+		{
+			if (m_delayCloseMenuTimer != null) 
+			{
+#if CMH_DEBUG
+				(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: CancelCloseSubMenu - Stopping m_delayCloseMenuTimer.", this));
+#endif // CMH_DEBUG
+
+				m_delayCloseMenuTimer.Stop();
+			}
+		}
+
+		internal void ClearStateFlags()
+		{
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: ClearStateFlags.", this));
+#endif // CMH_DEBUG
+
+			m_isPressed = false;
+			m_isPointerOver = false;
+			UpdateOwnerVisualState();
+		}
+
+		internal void OnIsEnabledChanged()
+		{
+			Control ownerAsControl = m_wpOwner?.Target as Control;
+
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnIsEnabledChanged - ownerAsControl=0x%p.", this, ownerAsControl));
+#endif // CMH_DEBUG
+
+			if (ownerAsControl != null)
+			{
+				bool bIsEnabled = false;
+				bIsEnabled = (ownerAsControl.IsEnabled);
+
+				if (!bIsEnabled)
+				{
+					ClearStateFlags();
+				}
+				else
+				{
+					ownerAsControl.UpdateVisualState(true /* useTransitions */);
+				}
+			}
+		}
+
+		public void OnVisibilityChanged()
+		{
+			UIElement ownerAsUIE = m_wpOwner?.Target as UIElement;
+
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnVisibilityChanged - ownerAsUIE=0x%p.", this, ownerAsUIE));
+#endif // CMH_DEBUG
+
+			if (ownerAsUIE != null)
+			{
+				Visibility visibility = Visibility.Collapsed;
+				visibility = ownerAsUIE.Visibility;
+
+				if (Visibility.Visible != visibility)
+				{
+					ClearStateFlags();
+				}
+			}
+		}
+
+		internal void OnPresenterSizeChanged(
+			object pSender,
+			SizeChangedEventArgs args,
+			Controls.Popup popup)
+		{
+			Control ownerAsControl = m_wpOwner?.Target as Control;
+
+			ISubMenuOwner ownerAsSubMenuOwner = m_wpOwner?.Target as ISubMenuOwner;
+
+			Control presenterAsControl = m_wpSubMenuPresenter?.Target as Control;
+
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPresenterSizeChanged - ownerAsControl=0x%p, ownerAsSubMenuOwner=0x%p.", this, ownerAsControl, ownerAsSubMenuOwner));
+#endif // CMH_DEBUG
+
+			if (ownerAsControl != null && ownerAsSubMenuOwner != null)
+			{
+				Size newPresenterSize = default;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'newPresenterSize'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)